lib/Ebook.php

```diff
@@ -472,7 +472,7 @@ protected function GetContributorsHtml(): string{
 				$this->_ContributorsHtml .= ' illustrated by ' . Ebook::GenerateContributorList($this->Illustrators, false) . ';';
 			}
 
-			if($this->_ContributorsHtml !== null){
+			if(!empty($this->_ContributorsHtml)){
```

Comment:
Is this correct? `empty()` checks if it's set and not false. But by convention protected variables accessed by getters and setters are null if unset. We should always check against null using `!==` and `===`.

Reply:
Line 462 above sets this variable to the empty string

```php
$this->_ContributorsHtml = '';
```

then appends to it if there are contributors, translators, or illustrators, so it can never be null on line 475.

I was looking for a clear way to check if the string was empty. Which of these options would read better?

```php
if($this->_ContributorsHtml != ''){
```

```php
if(strlen($this->_ContributorsHtml > 0)){
```

(This isn't a problem on the `master` branch because it uses a temporary variable, `$titleContributors`, to help set both `$ContributorsHtml` and `$TitleWithCreditsHtml`, so the null check on `$ContributorsHtml` works.)

https://github.com/standardebooks/web/blob/c6d2d77537d7e2e86a9646ba35db562406cbc856/lib/Ebook.php#L487